### PR TITLE
refactor logger & add json log format

### LINKED ISSUE #196: add json log format
We have added json log format to coreth. We can port those changes to Subnet-EVM ad well.

### PR DESCRIPTION
**Pull request by ceyonur**
Ports coreth log init and adds json log format. This one also removes MultiHandler with `vm.ctx.Log` since we don't support `io.Writer` in plugin context loggers. 

closes #196

### CODE REVIEW

File: plugin/evm/config.go [hunk 1]
```diff
@@ -35,6 +35,7 @@ const (
 	defaultPriorityRegossipTxsPerAddress          = 16
 	defaultOfflinePruningBloomFilterSize   uint64 = 512 // Default size (MB) for the offline pruner to use
 	defaultLogLevel                               = "info"
+	defaultLogJSONFormat                          = false
```
Comment (aaronbuchwald) [inline plugin/evm/config.go:38]:
```suggestion
	defaultLogJSONFormat                          = true
```

**Review by aaronbuchwald — APPROVED** [review]
LGTM - we should update to enable JSON logging by default

File: plugin/evm/config.go (hunk 1 above)
Comment (ceyonur) [reply plugin/evm/config.go:38]: coreth/avalanchego logs are not json by default. making this json by default would seem a little weird among others.

File: plugin/evm/vm.go [hunk 2]
```diff
@@ -243,14 +209,20 @@ func (vm *VM) Initialize(
...
 	}
 
-	// Set log level
-	logLevel, err := log.LvlFromString(vm.config.LogLevel)
+	vm.ctx = ctx
+
+	// Create logger
+	alias, err := vm.ctx.BCLookup.PrimaryAlias(vm.ctx.ChainID)
+	if err != nil {
+		alias = vm.ctx.ChainID.String()
```
Comment (darioush) [inline plugin/evm/vm.go:217]: can we log an error here (or is this expected to occur as a fallback and in that case can we clarify with a comment)?

**Review by darioush — APPROVED** [review]

File: plugin/evm/config.go (hunk 1 above)
Comment (aaronbuchwald) [reply plugin/evm/config.go:38]: Ah ok, I misunderstood something @StephenButtolph mentioned and thought there was a request to switch everything to use JSON logs by default.

File: plugin/evm/vm.go (hunk 2 above)
Comment (ceyonur) [reply plugin/evm/vm.go:218]: done

**Pull request merged**